opsramp/base.py

```diff
@@ -121,6 +122,62 @@ def chroot(self, suffix=''):
             self.tracker.reset()
         return self.compute_url()
 
+    @staticmethod
+    def collate_pages(get_request, data):
+        """Given a GET request whose results span across multiple pages, crawl
+        each page and collate the results.
+
+        :param first_page_data: "results" dict for first pagefull of dat
+        :type first_page_data: dict
+        :param request: Request used to get first page
+        :type request: requests.PreparedRequest
+        """
+        # First, sanity check that all is good. Only process GET requests:
+        if get_request.method.upper().strip() != "GET":
+            return data
+
```

Comment:
are there any GETs that are not paginated and therefore wouldnt expect this form of response?

Reply:
As far as I can see, yes there are, which we'd want to leave untouched.